ldap.go

```diff
@@ -33,15 +35,40 @@ func (c *LDAPConnection) Init(
 func (c *LDAPConnection) reconnectToLDAP() {
 	lcon, err := ldap.DialTLS("tcp", c.host,
 		&tls.Config{ServerName: "ldap.csh.rit.edu"})
-	if err != nil {
-		c.app.db.AddLog(0, "ldap connection error: "+err.Error())
-		log.Fatal(err)
+
+	maxReconnectionAttempts := 3 // TODO (wilnil): Use an environment variable?
+	attempts := 0
+
+	for err != nil {
+		if attempts < maxReconnectionAttempts {
+			c.app.db.AddLog(0, "ldap connection error: "+err.Error())
+			log.Info(err)
+			time.Sleep(2 * time.Second)
+			lcon, err = ldap.DialTLS("tcp", c.host,
+				&tls.Config{ServerName: "ldap.csh.rit.edu"})
+		} else {
+			c.app.db.AddLog(0, "ldap connection error: "+err.Error())
+			log.Fatal(err)
+		}
+		maxReconnectionAttempts++
 	}
+
 	err = lcon.Bind(c.bind_dn, c.bind_pw)
-	if err != nil {
-		c.app.db.AddLog(0, "ldap bind error: "+err.Error())
-		log.Fatal(err)
+
+	attempts = 0
+	for err != nil {
+		if attempts < maxReconnectionAttempts {
+			c.app.db.AddLog(0, "ldap bind error: "+err.Error())
+			log.Info(err)
```

Comment:
Dunno if log has this but if there's a log.Error you should use that so error logs are printed to stderr

Reply:
oh, brain, I totally didn't bother to look this up at 3AM